controllers/topics/index.js

```diff
@@ -170,6 +170,52 @@ const getTopics = async (req, res) => {
   }
 };
 
+const createTopic = async (req, res) => {
+  try {
+    let {name} = req.body;
+    name = name.toLowerCase();
+
+    const {token, userId} = req;
+
+    let topics = await Topics.findOne({
+      where: {
+        name,
+        deleted_at: null
+      }
+    });
+
+    if (topics) {
+      return res.status(400).json({
+        code: 400,
+        status: 'failed',
+        message: 'Topic already exists'
+      });
+    }
+
+    const topic_id = await insertTopics([name]);
+    await followTopicViaCreateTopic({token, name, topic_id, userId});
+
+    return res.status(200).json({
+      success: true,
+      message: 'Success',
+      topic_id
+    });
+  } catch (error) {
+    return res.json({
+      code: error.statusCode,
+      status: 'fail',
+      message: error.message,
+      data: 'null'
+    });
+  }
+};
+
+const followTopicViaCreateTopic = async ({token, name, topic_id, userId}) => {
+  const userTopicService = new UserTopicService(UserTopic, UserTopicHistory);
+  await userTopicService.followTopic(userId, topic_id);
+  await _afterPutTopic(false, token, userId, name, false, [], null, false);
+};
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

In the `followTopicViaCreateTopic` function, ensure that the `UserTopicService` and `_afterPutTopic` functions are properly imported and used. Also, consider adding error handling to this function.



<!-- This is an auto-generated comment by OSS CodeRabbit -->